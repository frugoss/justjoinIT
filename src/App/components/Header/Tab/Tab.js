import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './tab.module.scss';
import Button from '@material-ui/core/Button';

const Tab = ({ text, to, isActive, img }) => (
  <Link to={to} style={{textDecoration: "none"}}>
    <Button style={{textTransform:"none", marginBottom:4}} className={`${styles.tab} ${styles.tab} ${isActive ? styles.activeTab : ''}`} startIcon={img}>{text}</Button>
  </Link>
);

Tab.propTypes = {
  isActive: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  img: PropTypes.node.isRequired,
};


export default Tab;
