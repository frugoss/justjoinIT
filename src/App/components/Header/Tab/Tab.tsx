import React from 'react';
import { Link } from 'react-router-dom';
import styles from './tab.module.scss';
import Button from '@material-ui/core/Button';
import { makeStyles, createStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(createStyles({
  root: {
    textTransform:"none",
    marginBottom:4,
    width: 140
  }}));
type TabProps = {
  text: string,
  to: string,
  isActive: boolean,
  img: React.ReactNode
}
const Tab: React.FC<TabProps> = ({ text, to, isActive, img }) => {
  const buttonClass = useStyles();
  return (
      <Link to={to} className={styles.linkStyle}>
        <Button className={`${styles.tab} ${buttonClass.root} ${isActive ? styles.activeTab : ''}`} startIcon={img}>{text}</Button>
      </Link>
  );
}

export default Tab;
