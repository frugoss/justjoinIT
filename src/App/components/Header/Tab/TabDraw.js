import React from 'react';
import {Link} from 'react-router-dom';
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import { makeStyles } from '@material-ui/core/styles';


const useStyles = makeStyles({
  root: {
    color: "#B0BAC9",
    fontSize: 14,
    textDecoration: "none",
    fontWeight: 400,
    lineHeight: 1.5,
    width: "100%",
    alineHeight: 56,
    padding: "6px 6px 5px 0px",
  }
});

const TabDraw = ({to, text, img, setOpen}) => {
  const classy = useStyles();
  return (
      <>
        <Link onClick={() => setOpen(false)} to={to} style={{textDecoration: "none"}}>
          <ListItem style={{width:310}}>
            <ListItemIcon style={{marginLeft:5}}>
              {img}
            </ListItemIcon>
            <ListItemText  className={classy.root} primary={text} />
          </ListItem>
        </Link>
      </>
  );
};


export default TabDraw
