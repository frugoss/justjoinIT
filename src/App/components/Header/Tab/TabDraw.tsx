import React from 'react';
import {Link} from 'react-router-dom';
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import { makeStyles, createStyles } from '@material-ui/core/styles';


const useStyles = makeStyles(createStyles({
  root: {
    color: "#B0BAC9",
    fontSize: 14,
    textDecoration: "none",
    fontWeight: 400,
    lineHeight: 1.5,
    width: "100%",
    alineHeight: 56,
    padding: "6px 6px 5px 0px",
  }
}));
type TabDrawProps = {
  to: string,
  text: string,
  img: React.ReactNode,
  setOpen:  React.Dispatch<React.SetStateAction<boolean>>
}
const TabDraw: React.FC<TabDrawProps> = ({to, text, img, setOpen}) => {
  const classy = useStyles();
  return (
      <>
        <Link onClick={() => setOpen(false)} to={to} style={{textDecoration: "none"}}>
          <ListItem style={{width:310}}>
            <ListItemIcon style={{marginLeft:5}}>
              {img}
            </ListItemIcon>
            <ListItemText  className={classy.root} primary={text} />
          </ListItem>
        </Link>
      </>
  );
};


export default TabDraw
