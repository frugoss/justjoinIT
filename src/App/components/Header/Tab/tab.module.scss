@import '../../../style/variables';
@import '../../../style/breakpoints';

.tab {
  text-decoration: none;
  text-transform: none;
  line-height: 35px;
  font-weight: 500;
  &.activeTab, &:hover {
      color: rgb(244, 143, 177);
    }
}

