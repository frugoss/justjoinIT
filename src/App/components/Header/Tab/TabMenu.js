import WorkOutlineOutlinedIcon from "@material-ui/icons/WorkOutlineOutlined";
import BusinessOutlinedIcon from "@material-ui/icons/BusinessOutlined";
import React from "react";
import Tab from "./Tab";
import {faNewspaper} from '@fortawesome/free-solid-svg-icons';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faHandshake} from '@fortawesome/free-solid-svg-icons';
import TabOffers from "./TabOffers";


const tabs = [
    {
        to: '/ski-cams',
        text: 'Brand Stories',
        img: <BusinessOutlinedIcon/>,
    },
    {
        to: '/justgeekit',
        text: 'Just Geek IT',
        img: <FontAwesomeIcon icon={faNewspaper}/>,
    },
    {
        to: '/devs',
        text: 'Matchmaking',
        img: <FontAwesomeIcon icon={faHandshake}/>

    }
];


const TabMenu = ({location}) => {
    const isActive = destination => location.pathname === destination;
    return (
        <>
            <TabOffers location={location} to="/" text="Job Offers" img={<WorkOutlineOutlinedIcon/>}/>
            {tabs.map(({to, text, img}) => (
            <Tab
                img={img}
                key={to}
                to={to}
                text={text}
                isActive={isActive(to)}
            />

        ))
        }
        </>
    )
};

export default TabMenu;
