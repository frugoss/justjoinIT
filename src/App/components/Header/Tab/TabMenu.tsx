import WorkOutlineOutlinedIcon from "@material-ui/icons/WorkOutlineOutlined";
import BusinessOutlinedIcon from "@material-ui/icons/BusinessOutlined";
import React from "react";
import Tab from "./Tab";
import {faNewspaper} from '@fortawesome/free-solid-svg-icons';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faHandshake} from '@fortawesome/free-solid-svg-icons';
import TabOffers from "./TabOffers";
import History from 'history';


const tabs = [
    {
        to: '/ski-cams',
        text: 'Brand Stories',
        img: <BusinessOutlinedIcon/>,
    },
    {
        to: '/justgeekit',
        text: 'Just Geek IT',
        img: <FontAwesomeIcon icon={faNewspaper}/>,
    },
    {
        to: '/devs',
        text: 'Matchmaking',
        img: <FontAwesomeIcon icon={faHandshake}/>

    }
];

type TabMenuProps = {
    location: History.Location
}

const TabMenu: React.FC<TabMenuProps> = ({location}) => {
    const isActive = (destination: string) => location.pathname === destination;
    return (
        <>
            <TabOffers location={location} to="/" text="Job Offers" img={<WorkOutlineOutlinedIcon/>}/>
            {tabs.map(({to, text, img}) => (
            <Tab
                img={img}
                key={to}
                to={to}
                text={text}
                isActive={isActive(to)}
            />

        ))
        }
        </>
    )
};

export default TabMenu;
