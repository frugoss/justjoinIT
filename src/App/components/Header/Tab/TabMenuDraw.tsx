import WorkOutlineOutlinedIcon from "@material-ui/icons/WorkOutlineOutlined";
import BusinessOutlinedIcon from "@material-ui/icons/BusinessOutlined";
import React from "react";
import TabDraw from "./TabDraw";
import {faCode, faNewspaper} from '@fortawesome/free-solid-svg-icons';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faHandshake} from '@fortawesome/free-solid-svg-icons';
import List from "@material-ui/core/List";
import DrawerButton from "../Drawer/DrawerButton";
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import {userInterface} from '../../../utils/const'


type TabMenuDrawProps = {
    user: userInterface,
    logout: () => void,
    setOpen:  React.Dispatch<React.SetStateAction<boolean>>
}

const tabs = [
    {
        to: "/",
        text: "Job Offers",
        img: <WorkOutlineOutlinedIcon/>,
    },
    {
        to: '/ski-cams',
        text: 'Brand Stories',
        img: <BusinessOutlinedIcon/>,
    },
    {
        to: '/justgeekit',
        text: 'Just Geek IT',
        img: <FontAwesomeIcon icon={faNewspaper} size="lg"/>,
    },
    {
        to: '/devs',
        text: 'Matchmaking',
        img: <FontAwesomeIcon icon={faHandshake} size="lg"/>

    }
];


const TabMenuDraw: React.FC<TabMenuDrawProps> = ({user, logout, setOpen}) => {
    return (
        <>
            {user.auth ?
                    <DrawerButton setOpen={setOpen} to={"/dashboard"} text={"Dashboard"} img={<FontAwesomeIcon icon={faCode}/>}/>
             :<DrawerButton setOpen={setOpen} to={"/devs"} text={"Sign in"} img={
                <FontAwesomeIcon icon={faCode}/>}/>}
            <List> {tabs.map(({to, text, img}) => (
                <TabDraw
                    setOpen={setOpen}
                    img={img}
                    key={to}
                    to={to}
                    text={text}
                />

            ))
            }
                {user.auth ? <DrawerButton setOpen={setOpen} logout={logout} to={"/"} text={`Logout ${user && user.name ? user.name.slice(0, user.name.indexOf('@')): ""}`}  img={<ExitToAppIcon/>} />
                    : ""}
            </List>
        </>
    )

};

export default TabMenuDraw;
