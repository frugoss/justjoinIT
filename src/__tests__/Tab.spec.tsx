import React from 'react';
import "@testing-library/jest-dom/extend-expect"
import Tab, {TabProps} from "../App/components/Header/Tab/Tab";
import {render} from '@testing-library/react';
import {BrowserRouter} from "react-router-dom";


function renderTab(props: Partial<TabProps> = {}) {
    const defaultProps: TabProps = {
        text: "test text",
        img: "img",
        isActive: true,
        to: "test"

    };
    return render(<BrowserRouter><Tab {...defaultProps} {...props} /></BrowserRouter>);
}
describe("<Tab />", () => {
    it("snapshot",  () => {
        const {container} = renderTab();
        expect(container).toMatchSnapshot();

    })
})
