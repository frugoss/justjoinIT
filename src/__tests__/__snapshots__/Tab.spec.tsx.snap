// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab /> snapshot 1`] = `
<div>
  <a
    class="linkStyle"
    href="/test"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text tab makeStyles-root-1 activeTab"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          img
        </span>
        test text
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </a>
</div>
`;
